src/olympia/bandwagon/tests/test_views.py

```diff
@@ -1809,6 +1823,22 @@ def test_sorting(self):
             self.client.get(self.url + '?sort=-name'),
             self.addon_c, self.addon_b, self.addon_a)
 
+    def test_with_deleted_or_with_hidden(self):
+        response = self.send(self.url)
+        assert response.status_code == 200
+        # Normal
+        assert len(response.data['results']) == 3
+
+        response = self.send(self.url + '?filter=with_hidden')
+        assert response.status_code == 200
+        # Now there should be 2 extra
+        assert len(response.data['results']) == 5
+
+        response = self.send(self.url + '?filter=with_hidden_and_deleted')
+        assert response.status_code == 200
+        # And one more still - with_deleted gets you with_hidden too.
+        assert len(response.data['results']) == 6
+
```

Comment:
Can you add a quick test on the results to make sure we're returning the right ones ? No need to check for order, so just a `set()` with the ids is enough.

Reply:
https://github.com/mozilla/addons-server/pull/6855/commits/9d1a6eec308638821a644fe6205ef954f0910df2